HIP/hip-632.md

```diff
@@ -44,26 +44,46 @@ Additionally, to support standard implicit checks the network should support sys
   
 ## Specification
 
-A new system contract with functions to support address (Hedera Account ID and virtual) mapping reconciliation and authorization is proposed.
+A new system contract with functions to support address (Hedera Account ID and EVM address alias) mapping and authorization is proposed.
 
-This will aid developers to support smart contract migration logic for cases where Hedera Account ID or virtual address values are encountered. Additionally, EVM developers who were limited to `ECRECOVER`  authorization flows will be able to expand authorization checks to other key types. 
+This will aid developers to support smart contract migration logic for cases where aliases are encountered.
+Additionally, EVM developers, who were limited to `ECRECOVER` authorization flows, will be able to expand
+authorization checks to other key types. 
+
+HAS methods here can use EVM 20-byte addresses which are either Hedera account num aliases or EVM address 
+aliases. 
+
+* In this document the word _alias_, by itself, means either kind of alias.
+
+  (n.b.: See HIP-583 for definitions of Hedera account num alias and EVM address alias.  Hedera key aliases, defined there,
+are not supported by these methods.)
 
 | hash | signature | return | description |
 | --- | --- | --- | --- |
-|  | getVirtualAddresses(address) | address[] | returns an array of virtual addresses for a given Hedera Account ID  |
-|  | getHederaAddress(address) | address | returns the top level Hedera Account ID if applicable |
-|  | isVirtualAddress(address) | bool | true if valid virtual address, false if long-zero or non existing account |
-|  | isAuthorized(address, messageHash, signatureBlob) | bool | true if account is authorized to carry out transaction execution on account. Accepts protobuf key signature blobs. May be used for ECDSA, ED25519 and complex key flows |
-|  | isAuthorizedRaw(address, messageHash, signatureBlob) | bool | true if account is authorized to carry out transaction execution on account. Accepts single key raw signature blobs (ECDSA and ED25519). This provides similar logic to ECRECOVER. |
+| `0xdea3d081` | getEvmAddressAlias(address) | address | given a Hedera Account ID (account num alias) returns the EVM address alias, if present |
+| `0xbbf12d2e` | getHederaAccountNumAlias(address) | address | given an alias returns the top level Hedera Account ID (as account num alias), if applicable |
+| `0xcd82a665` | isAlias(address) | bool | true if an account num alias or evm address alias), false if non existing account |
```

Comment:
I think this should be isEvmAddressAlias()
```suggestion
| `0xcd82a665` | isEvmAddressAlias(address) | bool | true if a valid evm address alias on the network, false if an account num alias or non existing account |
```

Reply:
What we had was `isVirtualAddress` defined as `true if valid virtual address, false if long-zero or non existing account`.  

But we want to do two things at once:  Change the wording `virtual address` to `evm address alias` **and** remove the limitation "can't be a long-zero" which we felt was not a restriction we needed (or that I could find any reason for among us).  

So what we're left with, if we want to go that route, is "is evm address alias or long-zero _== account num alias_ and it must be for an existing account", i.e., is an alias and the account it is an alias for exists.  (Even hollow accounts.). Thus the name.

Do we still want to go in that direction (i.e., _also_ eliminate the restriction that it can't be a long-zero)?

Perhaps call it `isValidAlias` to make it explicit that it must be for a valid (== existing) account.